bika/lims/browser/analysisrequest/publish.py

```diff
@@ -1312,10 +1316,29 @@ def _analyses_data(self, ar, analysis_states=None):
                 andict['previous_results'] = ", ".join(
                     [p['formatted_result'] for p in andict['previous'][-5:]])
 
-            analyses.append(andict)
+            #Append addition analysis for each unit conversion
+            if andict['unit_conversions']:
+                for uc_uid in andict['unit_conversions']:
+                    new = dict(andict)
+                    unit_conversion = api.get_object_by_uid(
+                            uid=uc_uid, context=an)
+                    new['unit'] = unit_conversion.converted_unit
+                    new['formatted_unit'] = unit_conversion.converted_unit
+                    if andict.get('result'):
+                        new['formatted_result'] = new['result'] = \
+                            convert_unit(
+                                    andict['result'], 
+                                    unit_conversion.formula,
+                                    andict['obj'].getPrecision())
+                    analyses.append(new)
+            #Append original analysis
+            if not andict.get('hide_primary_result'):
+                analyses.append(andict)
```

Comment:
please refactor that to a more maintainable function, e.g.:

```python
def convert_unit_of(uid):
    obj = api.get_object_by_uid(uid)
    return {
        "unit": obj. converted_unit,
        "formatted_unit": obj.converted_unit,
         ...
    }
```

Then you can just do something like this:

```python
uc = convert_unit_of(uc_uid)
if uc:
    an_dict.update(uc)
```

Reply:
I refactored to make it more readable but not they way you did it. Maybe if you see what I did it will be clearer what is going on.

Reply:
Perfect, I'll review later :+1:
Of course, the purpose of my code snippets is just to give an impulse to show another way and just the first thing that comes into my mind. So it is absolute valid to discard it in favor of your own solution;)